.github/workflows/pre-release.yml

```diff
@@ -95,47 +83,75 @@ jobs:
       run: mv ./target/uberjar/metabase.jar bin/docker/.
     - name: Set up Docker Buildx
       id: buildx
-      uses: docker/setup-buildx-action@v1
+      uses: docker/setup-buildx-action@v2
       with:
         driver-opts: network=host
     - name: Build ${{ matrix.edition }} container
-      uses: docker/build-push-action@v2
+      uses: docker/build-push-action@v3
       with:
         context: bin/docker/.
         platforms: linux/amd64
         network: host
-        tags: localhost:5000/local-metabase:${{ env.IMAGE_TAG }}
+        tags: localhost:5000/local-metabase:${{ github.event.inputs.commit }}
         no-cache: true
         push: true
 
     - name: Launch container
-      run: docker run --rm -dp 3000:3000 localhost:5000/local-metabase:${{ env.IMAGE_TAG }}
+      run: docker run --rm -dp 3000:3000 localhost:5000/local-metabase:${{ github.event.inputs.commit }}
       timeout-minutes: 5
     - name: Wait for Metabase to start
       run: while ! curl -s 'http://localhost:3000/api/health' | grep '{"status":"ok"}'; do sleep 1; done
       timeout-minutes: 3
 
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKERHUB_RELEASE_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_RELEASE_TOKEN }}
     - name: Determine the target Docker Hub repository
       run: |
-        if [[ ${{ github.repository }} == 'metabase/metabase' ]]; then
-          if [[ ${{ matrix.edition }} == ee ]]; then
-            echo "Metabase EE: image ${{ env.IMAGE_TAG }} is going to be pushed to metabase/test-metabase-enterprise"
-            echo "DOCKERHUB_REPO=metabase/test-metabase-enterprise" >> $GITHUB_ENV
-          else
-            echo "Metabase OSS: image ${{ env.IMAGE_TAG }} is going to be pushed to metabase/test-metabase"
-            echo "DOCKERHUB_REPO=metabase/test-metabase" >> $GITHUB_ENV
-          fi
+        if [[ ${{ matrix.edition }} == ee ]]; then
+          echo "Metabase EE: image is going to be pushed to ${{ github.repository_owner }}/metabase-enterprise:${{ github.event.inputs.commit }}"
+          echo "DOCKERHUB_REPO=${{ github.repository_owner }}/metabase-enterprise:${{ github.event.inputs.commit }}" >> $GITHUB_ENV
```

Comment:
`github.repository_owner` is used here so that things are also working correctly in a fork (and therefore permits easy testing of this commit/PR).